### ISSUE TITLE
The `redundantNilInit` rule fails on properties in a struct

### ISSUE THREAD

**Issue by sindresorhus** [nicklockwood/SwiftFormat#875]
Swift supports both default implicit initialization of optional bindings and explicit.

These are equivalent:

```swift
struct A {
  var x: Int?
}
```

```swift
struct A {
  var x: Int? = nil
}
```

It would be nice to be able to enforce consistency here.

**Comment by nicklockwood** [comment]
SwiftFormat already has a `redundantNilInit` rule that removes the explicit `= nil` in these cases.

If you want to explicitly *add* the `= nil`, that would need a new configuration option to be added.

**Comment by sindresorhus** [comment]
Ok, so turns out it does work for:

```swift
var x: Int? = nil
```

but not:

```swift
struct A {
  var x: Int? = nil
}
```

So consider this a bug report.

**Comment by sindresorhus** [comment]
> If you want to explicitly add the = nil, that would need a new configuration option to be added.

No, I don't want that personally.

**Comment by nicklockwood** [comment]
> So consider this a bug report.

IIRC that's actually by design. It used to remove the nil in structs, but I changed it in response to a bug report that setting an explicit `nil` value makes that property optional in the default constructor, so removing the nil is a breaking change.

See: https://github.com/nicklockwood/SwiftFormat/issues/621

**Comment by sindresorhus** [comment]
It must have been fixed in a later Swift version then, because this compiles in Swift 5.3:

```swift
struct A {
  var x: Int? = nil
}

struct B {
  var x: Int?
}

A()
B()
```

**Comment by nicklockwood** [comment]
@sindresorhus interesting... I'm seeing the same. OK, I'll remove that limitation then.

**Comment by nicklockwood** [comment]
@sindresorhus fixed in 0.47.13
